tests/FunctionalTestCase.php

```diff
@@ -189,6 +196,45 @@ protected function assertCollectionExists(string $collectionName, ?string $datab
         }
     }
 
+    /** @param callable():void $closure */
+    protected static function commandResult(callable $closure): object
```

Comment:
Read command result from the events. That allows to keep the `assertCommandSucceeded`. But it might not be necessary if the operations throw an exception when the command fails.

Reply:
I concur. There should be no reason to maintain `assertCommandSucceeded()`, although removing it entirely might leave some tests without assertions (easy fix if so).

Reply:
I removed all that assertions.